### ISSUE TITLE
InaccessibleObjectException: Unable to make field protected boolean javax.swing.AbstractAction.enabled

### ISSUE THREAD

**Issue by don-vip** [jqno/equalsverifier#206]
# What steps will reproduce the problem?

Running `org.openstreetmap.josm.data.osm.search.SearchCompilerTest` with JDK 11 and `--illegal-access=deny`

# What is the code that triggers this problem?

```java
@Test
public void testEqualsContract() {
    Set<Class<? extends Match>> matchers = TestUtils.getJosmSubtypes(Match.class);
    for (Class<?> c : matchers) {
        EqualsVerifier.forClass(c).usingGetClass()
            .suppress(Warning.NONFINAL_FIELDS, Warning.INHERITED_DIRECTLY_FROM_OBJECT)
            .withPrefabValues(TaggingPreset.class, newTaggingPreset("foo"), newTaggingPreset("bar"))
            .verify();
    }
}

private static TaggingPreset newTaggingPreset(String name) {
    TaggingPreset result = new TaggingPreset();
    result.name = name;
    return result;
}
```

Provide an example of a complete class (equals method, hashCode method, relevant fields) and a call to EqualsVerifier.

```java
public class TaggingPreset extends javax.swing.AbstractAction {
    ....
    // equals and hashCode are not redefined in this class
}
```

# What error message or stack trace does EqualsVerifier give?

```
java.lang.reflect.InaccessibleObjectException: Unable to make field protected boolean javax.swing.AbstractAction.enabled accessible: module java.desktop does not "opens javax.swing" to unnamed module @67065cff
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:340)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:280)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:176)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:170)
	at nl.jqno.equalsverifier.internal.reflection.FieldAccessor.modify(FieldAccessor.java:177)
	at nl.jqno.equalsverifier.internal.reflection.FieldAccessor.copyTo(FieldAccessor.java:152)
	at nl.jqno.equalsverifier.internal.reflection.ObjectAccessor.copyInto(ObjectAccessor.java:116)
	at nl.jqno.equalsverifier.internal.reflection.ObjectAccessor.copy(ObjectAccessor.java:81)
	at nl.jqno.equalsverifier.EqualsVerifier.withPrefabValues(EqualsVerifier.java:137)
	at org.openstreetmap.josm.data.osm.search.SearchCompilerTest.testEqualsContract(SearchCompilerTest.java:721)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)
	at org.openstreetmap.josm.testutils.JOSMTestRules$TimeoutThread.run(JOSMTestRules.java:684)
```

# What did you expect?

No illegal access :)

# Which version of EqualsVerifier are you using?

2.5.1

# Please provide any additional information below.

**Comment by jqno** [comment]
Hi!

This is kind of a confusing issue, since I can't simply copy and paste the code into my IDE and run it to see what's wrong.

I see a mention of Swing in the stack trace. Where does that come from in the code?

**Comment by don-vip** [comment]
The `TaggingPreset` class we use in prefab values extends `javax.swing.AbstractAction`, I have edited the description to make it clearer.

**Comment by don-vip** [comment]
Standalone example:
```java
public class A {
    private B b;

    @Override
    public int hashCode() {
        return 31 + ((b == null) ? 0 : b.hashCode());
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null || getClass() != obj.getClass())
            return false;
        return java.util.Objects.equals(b, ((A) obj).b);
    }
}

public class B extends javax.swing.AbstractAction {

    public B(String name) {
        super(name);
    }

    @Override
    public void actionPerformed(java.awt.event.ActionEvent e) {
    }
}

public class Bug206 {
    @org.junit.Test
    public void testEqualsContract() {
        nl.jqno.equalsverifier.EqualsVerifier.forClass(A.class).usingGetClass()
            .suppress(
                    nl.jqno.equalsverifier.Warning.NONFINAL_FIELDS,
                    nl.jqno.equalsverifier.Warning.INHERITED_DIRECTLY_FROM_OBJECT)
            .withPrefabValues(B.class, new B("foo"), new B("bar"))
            .verify();
    }
}
```

**Comment by jqno** [comment]
Thanks for the update! I see the problem now, and I will look into it.

**Comment by jqno** [comment]
Sorry for the radio silence. I've reproduced the issue (which was pretty easy with your update). (See also [this repo](https://github.com/jqno/equalsverifier-issuerepro/tree/issue-206).)

I've been thinking about how to deal with this, but so far I can only conclude that I'm not sure yet 😅. Is this an urgent issue for you?

**Comment by don-vip** [comment]
Not urgent at all. Right now it is just a warning in our CI logs. I ran the test manually with `--illegal-access=deny` to help us diagnose the issue. You can take your time :)
It will only become urgent if a future version of Java changes the default behaviour to `--illegal-access=deny`. I don't think this will happen soon.

**Comment by jqno** [comment]
OK, thanks! :)
I'm currently working on a large new release, I think I'll look into this one once that release is out. I'll keep you posted.

**Comment by jqno** [comment]
It's fixed in version 3.1. You'll still get the illegal reflective access warnings if you run without `--illegal-access=deny`, but there will be no InaccessibleObjectException when you run with it.

**Comment by don-vip** [comment]
Thanks Santa! :santa: :)
